client/algod/wrappers.go

```diff
@@ -141,6 +142,21 @@ func (client Client) SuggestedParams(headers ...*Header) (response models.Transa
 	return
 }
 
+// BuildSuggestedParams gets the suggested transaction parameters and
```

Comment:
I think you should add to `SuggestedParams` instead of creating a new type. `LastRound` might be a little strange, but perhaps you can have `LastRound` and `LastRoundValid`? I guess you could also add `FirstRoundValid` and initialize it to the same value as `LastRound`.

Reply:
@winder I was thinking that leaving `models.TransactionParams` as exactly the struct returned from `algod`, and having a separate struct for the `Transaction` helpers, would make it clearer what `algod` is returning. What you propose is closer to the design document and python implementation though. What would the `json:` comments look like for these extra fields, do you know if there's an `omit` type flag?

Reply:
In the same vein, another goal was to keep the `SuggestedParams()` function returning the same object

Reply:
I guess whatever is most idiomatic is fine. You're right that the field names aren't perfect.

Keeping the original object and changing the constructor to something like this might be a better overall approach in the future:

https://algorand.atlassian.net/wiki/spaces/FDE/pages/491716666/SuggestedParams+Thoughts#Functional-Options

Reply:
@winder I think we'll move towards that with the indexer work (ex. an options struct using `url:` type options like `client.PendingTransaction` currently does, but I digress) for the transaction builders I wanted to hew close to the previous implementation and docs/keep the change smaller. I think this will be cleaned up more with the incoming indexer work, though.